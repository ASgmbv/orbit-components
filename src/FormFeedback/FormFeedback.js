// @flow
import * as React from "react";
import defaultTokens from "@kiwicom/orbit-design-tokens";
import styled from "styled-components";

export const TYPE_OPTIONS = {
  HELP: "help",
  ERROR: "error",
};

type Props = {
  children: React.Node,
  type: $Values<typeof TYPE_OPTIONS>,
  theme: typeof defaultTokens,
};

const StyledFormFeedback = styled(({ theme, type, ...props }) => <div {...props} />)`
  color: ${({ theme, type }) =>
    type === TYPE_OPTIONS.ERROR ? theme.colorTextError : theme.colorTextSecondary};
  font-family: ${({ theme }) => theme.fontFamily};
  font-size: ${({ theme }) => theme.fontSizeMessageForm}; // TODO: rename token
  font-weight: ${({ theme, type }) =>
    type === TYPE_OPTIONS.ERROR ? theme.fontWeightMedium : theme.fontWeightNormal};
  margin-top: 2px; // TODO: create token for 2px
  line-height: ${({ theme }) => theme.lineHeightText};

  & a {
    color: ${({ theme, type }) =>
      type === TYPE_OPTIONS.ERROR ? theme.colorTextError : theme.colorTextAttention};
    font-weight: ${({ theme }) => theme.fontWeightMedium};
    text-decoration: underline;
    cursor: pointer;
  }
`;

const FormFeedback = (props: Props) => {
  const { children, type, theme } = props;
  return (
    <StyledFormFeedback theme={theme} type={type}>
      {children}
    </StyledFormFeedback>
  );
};

FormFeedback.defaultProps = {
  type: "help",
  theme: defaultTokens,
};

export default FormFeedback;
