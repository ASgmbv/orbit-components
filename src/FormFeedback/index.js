// @flow
import * as React from "react";
import styled from "styled-components";

import defaultTokens from "../defaultTokens";
import TYPE_OPTIONS from "./consts";

import type { Props } from "./index";

const StyledFormFeedback = styled(({ theme, type, ...props }) => <div {...props} />)`
  color: ${({ theme, type }) =>
    type === TYPE_OPTIONS.ERROR ? theme.orbit.colorTextError : theme.orbit.colorTextSecondary};
  font-family: ${({ theme }) => theme.orbit.fontFamily};
  font-size: ${({ theme }) => theme.orbit.fontSizeMessageForm}; // TODO: rename token
  font-weight: ${({ theme, type }) =>
    type === TYPE_OPTIONS.ERROR ? theme.orbit.fontWeightMedium : theme.orbit.fontWeightNormal};
  margin-top: 2px; // TODO: create token for 2px
  line-height: ${({ theme }) => theme.orbit.lineHeightText};

  & a {
    color: ${({ theme, type }) =>
      type === TYPE_OPTIONS.ERROR ? theme.orbit.colorTextError : theme.orbit.colorTextAttention};
    font-weight: ${({ theme }) => theme.orbit.fontWeightMedium};
    text-decoration: underline;
    cursor: pointer;
  }
`;

StyledFormFeedback.defaultProps = {
  theme: defaultTokens,
};

const FormFeedback = (props: Props) => {
  const { children, type = TYPE_OPTIONS.HELP } = props;
  return <StyledFormFeedback type={type}>{children}</StyledFormFeedback>;
};

export default FormFeedback;
