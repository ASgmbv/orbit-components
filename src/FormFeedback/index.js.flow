// @flow
import type { ReactComponentStyled } from "styled-components";

import type { Globals, UseRef } from "../common/common.js.flow";

export type Props = {
  +error?: React$Node,
  +help?: React$Node,
  +tooltipShown?: boolean,
  +tooltipShownHover?: boolean,
  +labelRef: UseRef,
  +iconRef: UseRef,
  ...Globals,
};

declare export default React$ComponentType<Props>;

declare export var StyledFormFeedback: ReactComponentStyled<any>;
