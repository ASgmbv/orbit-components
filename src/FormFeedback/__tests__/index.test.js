// @flow
import * as React from "react";
import { shallow } from "enzyme";

import FormFeedback from "../index";

describe("FormFeedback", () => {
  const dataTest = "test";
  const error = "err";
  const component = shallow(
    <FormFeedback tooltipShown error={error} dataTest={dataTest}>
      FormFeedback
    </FormFeedback>,
  );
  it("should have data-test", () => {
    expect(component.find("FormFeedbackTooltip").prop("dataTest")).toBe(dataTest);
  });
  it("should match snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
