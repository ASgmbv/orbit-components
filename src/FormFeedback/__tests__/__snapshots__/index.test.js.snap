// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFeedback should match snapshot 1`] = `
<Fragment>
  <FormFeedbackTooltip
    dataTest="test"
    shown={true}
  >
    err
  </FormFeedbackTooltip>
</Fragment>
`;
